File: frontend/src/types/task.ts
// frontend/src/types/task.ts

export interface TaskResponse {
  id: string;          // UUID
  user_id: string;     // Owner identifier
  title: string;       // Task title (1-255 chars)
  description: string | null;  // Optional description
  is_completed: boolean;       // Completion status
  priority: number;    // Priority level (1-5)
  created_at: string;  // ISO datetime string
  updated_at: string;  // ISO datetime string
}

export interface TaskCreate {
  title: string;       // Required (1-255 chars)
  description?: string | null;  // Optional
  priority?: number;   // Optional (default: 1)
  is_completed?: boolean;       // Optional (default: false)
}

export interface TaskUpdate {
  title?: string;      // Optional update
  description?: string | null;  // Optional update
  priority?: number;   // Optional update
  is_completed?: boolean;       // Optional update
}

File: backend/app/schemas/task.py
from pydantic import BaseModel, ConfigDict, Field
from typing import Optional
from uuid import UUID
from datetime import datetime


class TaskCreate(BaseModel):
    """Schema for creating a new task."""
    title: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = None
    priority: int = Field(default=1, ge=1, le=5)
    is_completed: bool = False


class TaskUpdate(BaseModel):
    """Schema for updating an existing task."""
    model_config = ConfigDict(from_attributes=True)
    title: Optional[str] = Field(default=None, min_length=1, max_length=255)
    description: Optional[str] = None
    is_completed: Optional[bool] = None
    priority: Optional[int] = Field(default=None, ge=1, le=5)


class TaskResponse(BaseModel):
    """Schema for task response."""
    model_config = ConfigDict(from_attributes=True)
    id: UUID
    user_id: str
    title: str
    description: Optional[str] = None
    is_completed: bool
    priority: int
    created_at: datetime
    updated_at: datetime
